core/src/main/java/net/opentsdb/aura/metrics/core/TimeSeriesShard.java

```diff
@@ -606,7 +635,22 @@ public Job createNext() {
 
     @Override
     public void close() {
-      //Do nothing
+      if (tsTableClone != null) {
+        if (!isPurgeDone.compareAndSet(false, true)) {
+          if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug("Purge was already marked finished?",
+                    new RuntimeException("Call stack"));
+          }
+        } else {
+          tsTableClone.close();
+          tsTableClone = null;
+          new RuntimeException().printStackTrace();
```

Comment:
??

Reply:
what is `new RuntimeException().printStackTrace()` for

Reply:
doh.... debugging flub.